src/kernel-install/kernel-install

```diff
@@ -90,7 +90,13 @@ if [[ ! $COMMAND ]] || [[ ! $KERNEL_VERSION ]]; then
     exit 1
 fi
 
+# Prefer to use an existing machine ID from /etc/machine-info or /etc/machine-id. Otherwise, try to generate
+# a new machine ID in /etc/machine-info. If that fails, use "Default".
+
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && MACHINE_ID="$(source /etc/machine-info && echo $KERNEL_INSTALL_MACHINE_ID)"
 [ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && [ "$(stat -fc %T /etc/machine-id)" != "tmpfs" ] && read -r MACHINE_ID < /etc/machine-id
```

Comment:
As discussed, please don't check for tmpfs here. Also, I don't think we should create a new machine-id. Is there any reason not to use the volatile machine-id? It might be volatile right now, but something can save it to disk later.

Reply:
> . Also, I don't think we should create a new machine-id. 

We don't create a new machine-id do we? We just generate something to put in machine-info when no machine ID is available yet.

> Is there any reason not to use the volatile machine-id? It might be volatile right now, but something can save it to disk later.

Sure, but if there isn't, you end up with a new directory in $BOOT every time you run kernel-install which doesn't seem ideal. I don't think users would ever really intend to use a temp machine ID as the ID in $BOOT. In practice we'd fall back to "Default" if MACHINE_ID is tmpfs since it's very likely that means /etc/ is read-only so writing a new ID to machine-info will fail and we'll fall back to "Default". That seems better than creating a new directory every time since we can properly remove old versions that way.

Ideally in all these cases, machine-info will contain KERNEL_INSTALL_MACHINE_ID though so we never even have to check /etc/machine-id or fall back to "Default".

Reply:
My idea was that we take whatever machine-id is already set up (hence my ask to remote the check for tmpfs), and "freeze" it by writing it to /etc/machine-info.

Something like this:
```bash
do_write=0
[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && MACHINE_ID="$(source /etc/machine-info && echo $KERNEL_INSTALL_MACHINE_ID)"
[ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && read -r MACHINE_ID </etc/machine-id && do_write=1
[ -z "$MACHINE_ID" ] && "KERNEL_INSTALL_MACHINE_ID=$(systemd-id128 new)" && do_write=1
[ $do_write -eq 1 ] && echo "KERNEL_INSTALL_MACHINE_ID=$MACHINE_ID" >>/etc/machine-info
```